### PR TITLE
System.cpp: Do not require posix_spawn on "Linux"

### PR DESCRIPTION
**Pull request by glebm**
Many libc implementations do not implement posix_spawn.
Allow love2d to build without it.

### CODE REVIEW

File: src/modules/system/System.cpp [hunk 1]
```diff
@@ -104,10 +117,18 @@ bool System::openURL(const std::string &url) const
...
 	const char *argv[] = {"xdg-open", url.c_str(), nullptr};
 
+#ifdef LOVE_HAS_POSIX_SPAWN
 	// Note: at the moment this process inherits our file descriptors.
 	// Note: the below const_cast is really ugly as well.
 	if (posix_spawnp(&pid, "xdg-open", nullptr, nullptr, const_cast<char **>(argv), environ) != 0)
 		return false;
+#else
+	pid = fork();
+	if (pid == 0) {
```
Comment (slime73) [inline src/modules/system/System.cpp:127]: The `{` should be in its own line, as per https://love2d.org/wiki/Code_Style

File: src/modules/system/System.cpp [hunk 2]
```diff
@@ -39,7 +39,18 @@
 #if defined(LOVE_ANDROID)
 #include "common/android.h"
 #elif defined(LOVE_LINUX)
+
+#ifdef __has_include
```
Comment (slime73) [inline src/modules/system/System.cpp:43]: What's the availability of `__has_include` like? Is it possible for `posix_spawnp` to exist but `__has_include` isn't supported?

**Comment by slime73** [comment]
> Many libc implementations do not implement posix_spawn.

Which Linux systems in particular that love would otherwise would build on are prevented from being supported by the current `posix_spawn` code?

File: src/modules/system/System.cpp (hunk 2 above)
Comment (glebm) [reply src/modules/system/System.cpp:43]: Yes, they're orthogonal: `__has_include` is a C++17 language feature, `posix_spawn` is a function available in some libc implementations.

**Comment by glebm** [comment]
> Which Linux systems in particular that love would otherwise would build on are prevented from being supported by the current `posix_spawn` code?

All of the handheld devices that run OpenDingux-like systems. Example device: RetroGame RG350P

They use versions of uClibc(-ng) as their libc and it doesn't have `posix_spawnp`.

File: src/modules/system/System.cpp (hunk 2 above)
Comment (glebm) [reply src/modules/system/System.cpp:43]: clang and gcc have had `__has_include` for a long time (way before it was standardized in C++17)

**Comment by glebm** [comment]
Another big target is OpenWRT router firmware (also uClibc). Doesn't everyone want to run love2d games on their router? :smiley:

**Comment by MikuAuahDark** [comment]
While this change is _probably_ okay, running LOVE at platforms you mentioned may be hard, if not impossible because those platforms most likely lack the power and memory to run LOVE.

**Comment by glebm** [comment]
@MikuAuahDark That's not true, these devices have been using an older version of Love since 2014!

RG350 has 512MiB of RAM and a GPU

File: src/modules/system/System.cpp (hunk 1 above)
Comment (glebm) [reply src/modules/system/System.cpp]: Done

**Pull request merged**